src/constants.ts

```diff
@@ -19,6 +20,9 @@ const metaflowServiceUrl = new URL(
   window.METAFLOW_SERVICE || process.env.REACT_APP_METAFLOW_SERVICE || '/api',
   document.baseURI,
 );
+
+export const appBasePath = window.APP_BASE_PATH || process.env.REACT_APP_BASE_PATH;
```

Comment:
?? rather than || would be better . In the case where `window.APP_BASE_PATH` was deliberately set to "", || would set `appBasePath` to `process.env.REACT_APP_BASE_PATH` which may be undefined whereas ?? would set `appBasePath` to ""